File: lesson4/c_c++/homework/homework4_16.cpp
//: lesson4/c_c++/homework/homework4_16.cpp
#include <iostream>

int main() {
    int entered_numb;
    std::cout << "Введите натуральное число => ";
    std::cin >> entered_numb;

    if (entered_numb > 0 && entered_numb < 100) {
        int result1 = entered_numb % 10;
        int result2 = entered_numb / 10 % 10;
        if (result1 == result2) {
            std::cout << "Оба числа одинаковы.\n";
        } else {
            std::cout << "Числа неодинаковы.\n";
        }
    } else if (entered_numb > 100 && entered_numb < 1000) {
        int result1 = entered_numb % 10;
        int result2 = entered_numb / 10 % 10;
        int result3 = entered_numb / 100;
        if (result1 == result2 && result3 == result1) {
            std::cout << "Все числа одинаковы.\n";
        } else {
            std::cout << "Числа неодинаковы.\n";
        }
    } else {
        int result1 = entered_numb % 10;
        int result2 = entered_numb / 10 % 10;
        int result3 = entered_numb / 100 % 10;
        int result4 = entered_numb / 1000;
        if (result1 == result2 && result3 == result4 && result1 == result3 && result2 == result4) {
            std::cout << "Все числа одинаковы.\n";
        } else {
            std::cout << "Числа неодинаковы.\n";
        }
    }
    return 0;
}

File: lesson4/python3/homework/homework4_16.py
#!/usr/bin/python3
#: lesson4/python3/homework/homework4_16.py

entered_numb = int(input("Введите натуральное число => "))

result1 = entered_numb % 10
result2 = entered_numb // 10 % 10
result3 = entered_numb // 100
result4 = entered_numb // 100 % 10
result5 = entered_numb // 1000
if entered_numb > 0 and entered_numb < 100:
    if result1 == result2:
        print("Оба числа одинаковы.")
    else:
        print("Числа неодиноковы.")
elif entered_numb > 100 and entered_numb < 1000:
    if result1 == result2 and result3 == result1:
        print("Все числа одинаковы.")
    else:
        print("Числа неодинаковы.")
else:
    if result1 == result2 and result4 == result5 and result1 == result5:
        print("Все числоа одинаковы.")
    else:
        print("Числа неодинаковы.")

